### PR TITLE
[MO] Add check for Python version to NetworkX dependency

### PR DESCRIPTION
**Pull request by achetver**
### Details
Update NetworkX dependency check for MO and POT.

### Tickets:
 - 72806

### CODE REVIEW

**Review by popovaan — APPROVED** [review]

**Comment by achetver** [comment]
@pavel-esir @sadolini please, review these changes.

**Review by sadolini — APPROVED** [review]

**Review by pavel-esir — APPROVED** [review]

File: tools/mo/requirements_tf.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # TensorFlow 1.x and 2.x are incompatible, use separate virtual environments for each version
 tensorflow~=1.15.5
-networkx~=2.5
+networkx~=2.5; python_version == "3.6"
```
Comment (rkazants) [inline tools/mo/requirements_tf.txt:3]: I think we should install ~=2.5 version for earlier python version despite we do not support it 3.5 or earlier versions.

Comment (achetver) [reply tools/mo/requirements_tf.txt]: Yes, we should.
Updated, now we check python_version <= "3.6"

File: tools/mo/requirements_tf2.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 tensorflow~=2.5.0
-networkx~=2.5
+networkx~=2.5; python_version <= "3.6"
+networkx~=2.6; python_version > "3.6"
```
Comment (rkazants) [inline tools/mo/requirements_tf2.txt:3]: We also have other places still using 2.5 openvino\tools\openvino_dev\requirements.txt (and thirdparty/open-model-zoo).
@slyubimt, could you please recommend the correct way to change networkx version so we willn't affect other componenets

**Comment by rkazants** [comment]
@slyubimt, could you please take a look at this change?

File: tools/mo/requirements_tf2.txt (hunk 2 above)
Comment (slyubimt) [reply tools/mo/requirements_tf2.txt:3]: Currently `openvino-dev` packaging script obtains requirements from setup.py of individual component. 
This change needs to be agreed  with components owners and aligned for components to void potential conflicts:
 - POT @AlexKoff88 :  https://github.com/openvinotoolkit/openvino/blob/5f53e6f697038d454f29777c81afe05020cdbec1/tools/pot/setup.py#L52

Comment (achetver) [reply tools/mo/requirements_tf2.txt:3]: I update NetworkX dependency in POT to align it with MO dependency.

@AlexKoff88 @KodiaqQ Are you ok with that?

Comment (achetver) [reply tools/mo/requirements_tf2.txt:3]: POT requirements list format was aligned with https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#platform-specific-dependencies

Comment (achetver) [reply tools/mo/requirements_tf2.txt:3]: POT CI check is green. MO conversion checked with NetworkX 2.6 and found no issues.

**Review by rkazants — APPROVED** [review]

**Comment by rkazants** [comment]
@achetver, please resolve conflicts created after Deprecation of TF1 merge

**Comment by achetver** [comment]
> @achetver, please resolve conflicts created after Deprecation of TF1 merge

Done.

**Review by rkazants — APPROVED** [review]

**Pull request merged**